src/N3Lexer.js

```diff
@@ -302,12 +302,20 @@ export default class N3Lexer {
       case ')':
       case '{':
       case '}':
+        if (input.length > 1 && input[1] === '|') {
+          type = '{|', matchLength = 2;
+          break;
+        }
```

Comment:
…but that's only for `}`, right?

So we'd need
```suggestion
      case '{':
        if (input.length > 1 && input[1] === '|') {
          type = '{|', matchLength = 2;
          break;
        }
      case '{':
      case '[':
      case ']':
      case '(':
      case ')':
      case ',':
      case ';':
```

Reply:
Nah, even worse:
```suggestion
      case '{':
        if (input.length < 2) {
          inconclusive = true;
        }
        if (input[1] === '|')
          type = '{|', matchLength = 2;
          break;
        }
      case '{':
      case '[':
      case ']':
      case '(':
      case ')':
      case ',':
      case ';':
```

and have a test where the input stream breaks at `{` and the next chunk starts with `|`.